tests/e2e/grpc/grpc_test.go

```diff
@@ -0,0 +1,205 @@
+package grpc
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/jexia/semaphore"
+	"github.com/jexia/semaphore/cmd/semaphore/daemon/providers"
+	codecProto "github.com/jexia/semaphore/pkg/codec/proto"
+	"github.com/jexia/semaphore/pkg/specs"
+	transportGRPC "github.com/jexia/semaphore/pkg/transport/grpc"
+	"github.com/jexia/semaphore/tests/e2e"
+	proto "github.com/jexia/semaphore/tests/e2e/grpc/proto"
+	"google.golang.org/grpc"
+)
+
+const (
+	SemaphorePort = 5051
+	SemaphoreHost = "localhost"
+)
+
+var SemaphoreGRPCAddr = fmt.Sprintf("%s:%d", SemaphoreHost, SemaphorePort)
+
+// convert any interface{} to proto struct and back (use with recursive types).
+func convert(t *testing.T, src, dst interface{}) {
+	encoded, err := json.Marshal(src)
+	if err != nil {
+		t.Fatalf("unable to encode: %s", err)
+	}
+
+	if err := json.Unmarshal(encoded, dst); err != nil {
+		t.Fatalf("unable to encode: %s", err)
+	}
+}
+
+type external struct {
+	*proto.UnimplementedExternalServer
+}
+
+func (*external) Post(_ context.Context, input *proto.Data) (*proto.Data, error) {
+	return input, nil
+}
+
+func TestGRPCTransport(t *testing.T) {
```

Comment:
Please add a short flag check to skip integration tests. Unexpected errors will now be thrown when an external service is running on one of the hardcoded ports.